src/pip/_internal/commands/show.py

```diff
@@ -116,10 +116,14 @@ def _get_requiring_packages(current_dist: BaseDistribution) -> Iterator[str]:
 
         metadata = dist.metadata
 
+        location = dist.location or ""
+        if dist.editable:
+            location = str(dist.editable_project_location)
```

Comment:
Since `editable_project_location` would be None if the dist is not editable, this is likely easier?

```python
location = dist.editable_project_location or dist.location or ""
```

Reply:
agreed. changed :)